include/nudb/basic_store.hpp

```diff
@@ -343,6 +361,61 @@ class basic_store
         error_code& ec,
         Args&&... args);
 
+    /** Open a database.
+
+        The database identified by the specified directory
+        under which default data and key paths are opened.
+
+        @par Requirements
+
+        The database must be not be open.
+
+        @par Thread safety
+
+        Not thread safe. The caller is responsible for
+        ensuring that no other member functions are
+        called concurrently.
+
+        @param dir_path The path to the directory containing data
+        and key files.
+
+        @param ec Set to the error, if any occurred.
+
+        @param args Optional arguments passed to @b File constructors.
+
+    */
+    template<class... Args>
+    void
+    open(
```

Comment:
This really feels like it should be a free function which calls into the existing `open`  member function. The function could be `open_dir`.